packages/compiler-core/src/transform.ts

```diff
@@ -250,7 +250,11 @@ export function createTransformContext(
       }
     },
     hoist(exp) {
-      context.hoists.push(exp)
+      const index = context.hoists.findIndex(
+        d => d && d.loc.source === exp.loc.source
+      )
+      if (index === -1) context.hoists.push(exp)
```

Comment:
Can't we simply use `some` here, if we just want to check that element doesn't exist in array?

```suggestion
      const exists = context.hoists.some(el => el && el.loc.source === exp.loc.source)
      if (!exists) context.hoists.push(exp)
```

Reply:
I think `findIndex `is faster than `some`

see https://www.measurethat.net/Benchmarks/Show/6749/0/some-vs-filter-vs-findindex